src/components/ChainConnection.tsx

```diff
@@ -16,21 +16,57 @@ import {
   chainConnectionAtom,
   networkConfigAtom,
   termsIndexAgreedUponAtom,
+  smartWalletProvisionedAtom,
+  provisionToastIdAtom,
+  ChainConnection as ChainConnectionStore,
 } from 'store/app';
-import { watchContract, watchPurses } from 'utils/updates';
+import {
+  watchContract,
+  watchPurses,
+  watchSmartWalletProvision,
+} from 'utils/updates';
 import NetworkDropdown from 'components/NetworkDropdown';
 import TermsDialog, { currentTermsIndex } from './TermsDialog';
 import clsx from 'clsx';
 import { makeAgoricChainStorageWatcher } from '@agoric/rpc';
 import { sample } from 'lodash-es';
-import { makeImportContext } from '@agoric/smart-wallet/src/marshal-contexts';
+import { loadNetworkConfig } from 'utils/networkConfig';
+import ProvisionSmartWalletDialog from './ProvisionSmartWalletDialog';
 
 import 'react-toastify/dist/ReactToastify.css';
 import 'styles/globals.css';
-import { loadNetworkConfig } from 'utils/networkConfig';
+import { querySwingsetParams } from 'utils/swingsetParams';
 
 const autoCloseDelayMs = 7000;
 
+const useSmartWalletFeeQuery = (
+  chainConnection: ChainConnectionStore | null
+) => {
+  const [smartWalletFee, setFee] = useState<bigint | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    const fetchParams = async () => {
+      assert(chainConnection);
+      try {
+        const params = await querySwingsetParams(
+          chainConnection.watcher.rpcAddr
+        );
+        console.debug('swingset params', params);
+        setFee(BigInt(params.params.powerFlagFees[0].fee[0].amount));
```

Comment:
Another use case for `theOne`. Let's look for these through the dapps once we have that.

Reply:
In this instance at least we handle the error gracefully on line 103